src/app/manager-dashboard/reports/reports-detail.component.ts

```diff
@@ -285,8 +287,8 @@ export class ReportsDetailComponent implements OnInit, OnDestroy {
       }
     ];
     const formGroup = {
-      startDate: new Date(minDate),
-      endDate: [ new Date(this.today), CustomValidators.endDateValidator() ]
+      startDate: new Date(startDateFromForm.setHours(0, 0, 0, 0)),

```

Comment:
You should remove `setHours` and you do not need to create a `new Date` here.  This also applies to `endDate`